### ISSUE TITLE
Scraper priority suggestion

### ISSUE THREAD

**Issue by icpryde** [nixgates/plugin.video.seren#67]
I'd like to suggest an improvement to scraping, currently the method for scraping is lets say we have 2 packages lets call them one and 2. Currently it searches through all of them providers and once it finds 
 2 files the "pre-emptive termination" will kick in and then load the file. 

The issue with this is sometimes it will scrape 10 - 30 files from the one package before it starts to load the ones we want from the other package.

My suggestion would be to separate the process so you can make it so it only searches in the one package and if it finds something then loads as it would, only if it can't locate anything it then goes looking into the other scraper package to find a file.

in other words set priorities for packages so it scrapes the one and only uses the other if it cant locate something.

**Comment by nixgates** [comment]
I have an idea on how to handle automated scraper priority that I brought up with the test group a little while ago.

I'm hoping to get moving on it and put into testing soon. It's going to take a fair bit of tinkering to get up and running properly but it's coming.

I'll give more details on how it works and prioritises scrapers soon.

**Comment by icpryde** [comment]
That’s awesome to hear :)

**Comment by nixgates** [comment]
@icpryde I was just reading through this ticket again to see what needs to be done. 

Just so you are aware, Seren doesn't go through providers one by one, they are all performed at the same time. What you are likely seeing is a provider finishing before the other. 

My ideas for scraper priority are probably not going to happen at this point due to there really not call for it. If that changes in the future I'll work on it but until then, likely not.

What I suggest you do is adjust your pre-emptive termination setting up or turn it off if there are providers not completing when you want.
